File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int buyPrice = prices[0];
        int profit = 0;
            
        foreach(int p in prices){
            buyPrice = Math.Min(buyPrice, p);
            profit = Math.Max(profit, p-buyPrice);
        }
        
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let maxProfit: number = 0;
    let minPrice: number = Number.MAX_VALUE;
    
    for(let i=0; i < prices.length; i++){
        if(minPrice > prices[i]){
            minPrice = prices[i];
        } else if(prices[i] - minPrice > maxProfit){
            maxProfit = prices[i] - minPrice;
        }            
    }
    return maxProfit;
};